File: src/ios/WebViewAutoScroll.m
//
//  WebViewAutoScroll.m
//  chameleon
//
//  Created by Foreveross on 16/8/30.
//
//

#import "WebViewAutoScroll.h"
#import "AppDelegate.h"

static BOOL isInvalidated = NO;
//false
@interface WebViewAutoScroll()<UIScrollViewDelegate>

@end

@implementation WebViewAutoScroll

- (void)enable:(CDVInvokedUrlCommand *)command{
    if (!isInvalidated) {
        [AppDelegate instance].viewController.webView.scrollView.delegate = nil;
    }
}

- (void)disable:(CDVInvokedUrlCommand *)command{
    if (!isInvalidated) {
        [AppDelegate instance].viewController.webView.scrollView.delegate = self;
    }
}

- (void)invalidate:(CDVInvokedUrlCommand *)command{
    [AppDelegate instance].viewController.webView.scrollView.delegate = self;
    isInvalidated = YES;
}
    
- (UIView *)viewForZoomingInScrollView:(UIScrollView *)scrollView{
    return nil;
}

@end

File: www/webViewAutoScroll.js
       var exec = require('cordova/exec');
       module.exports = {
       enable : function() {
               exec(null, null, "WebViewAutoScroll", "enable", []);
       },
       disable:function() {
               exec(null, null, "WebViewAutoScroll", "disable", []);
       },
       invalidate:function() {
       exec(null, null, "WebViewAutoScroll", "invalidate", []);
       },
   };
